pcsx2-qt/MainWindow.cpp

```diff
@@ -1052,12 +1088,21 @@ bool MainWindow::requestShutdown(bool allow_confirm, bool allow_save_to_state, b
 	// If we don't have a crc, we can't save state.
 	allow_save_to_state &= (m_current_disc_crc != 0);
 	bool save_state = allow_save_to_state && default_save_to_state;
+	VMLock lock(pauseAndLockVM());
 
+	// Redundant initial check of IsBusy() to prevent hitting no on this prompt,
+	// from then bringing up the normal shutdown prompt too.
+	if (MemcardBusy::IsBusy())
+	{
+		// Check if memcard is busy, deny request if so.
+		if (shouldAbortForMemcardBusy(lock))
+		{
+			return false;
+		}
+	}
 	// Only confirm on UI thread because we need to display a msgbox.
-	if (!m_is_closing && allow_confirm && !GSDumpReplayer::IsReplayingDump() && Host::GetBoolSettingValue("UI", "ConfirmShutdown", true))
+	else if (!m_is_closing && allow_confirm && !GSDumpReplayer::IsReplayingDump() && Host::GetBoolSettingValue("UI", "ConfirmShutdown", true))
```

Comment:
Not sure if this should be else. It's skipping the confirm dialog, which means there's no opportunity save the resume state.